price-feeder/oracle/provider/okx.go

```diff
@@ -158,9 +186,21 @@ func (p *OkxProvider) messageReceived(messageType int, bz []byte) {
 	}
 
 	var tickerResp OkxTickerResponse
+	var candleResp OkxCandleResponse
+
+	// sometimes it returns other messages which are not
+	// tickerResponses or candleResponses
 	if err := json.Unmarshal(bz, &tickerResp); err != nil {
-		// sometimes it returns other messages which are not tickerResponses
-		p.logger.Err(err).Msg("could not unmarshal")
+		if err := json.Unmarshal(bz, &candleResp); err != nil {
```

Comment:
Does the placement of this `json.Unmarshal(bz, &candleResp)` inside of the error case cause the candle unmarshal to be missed for candle responses?

If I recall correctly, unmarshaling json of struct A into struct B will ignore the fields of A that are not present in B, without returning an error.

[sauce](https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/encoding/json/decode.go)
```go
// To unmarshal JSON into a struct, Unmarshal matches incoming object
// keys to the keys used by Marshal (either the struct field name or its tag),
// preferring an exact match but also accepting a case-insensitive match. By
// default, object keys which don't have a corresponding struct field are
// ignored (see Decoder.DisallowUnknownFields for an alternative).
```

Reply:
Nope, the error you get from the first attempt is due to a type disagreement : 
```
error(*encoding/json.UnmarshalTypeError) *{Value: "array", Type: reflect.Type(*reflect.rtype) *{size: 48, ptrdata: 40, hash: 1242516269, tflag: tflagUncommon|tflagExtraStar|tflagNamed (7), align: 8, fieldAlign: 8, kind: 25, equal: type..eq.github.com/umee-network/umee/price-feeder/oracle/provider.OkxTickerPair, gcdata: *21, str: 247223, ptrToThis: 1015424}, Offset: 60, Struct: "OkxTickerResponse", Field: "data"}
```
This isn't a super clear way to do it, maybe it'd be better to do : 
```golang
	// sometimes it returns other messages which are not
	// tickerResponses or candleResponses
	if err := json.Unmarshal(bz, &tickerResp); err != nil {
		p.logger.Debug().Err(err).Msg("could not unmarshal ticker resp")
	}
	if err := json.Unmarshal(bz, &candleResp); err != nil {
		p.logger.Debug().Err(err).Msg("could not unmarshal candle resp")
	}

	for _, tickerPair := range tickerResp.Data {
		p.setTickerPair(tickerPair)
	}
	for _, candlePair := range candleResp.Data {
		p.setCandlePair(candlePair, candleResp.ID.InstID)
	}
```

a bit less efficient, but way more readable